### PR TITLE
Fix compiler warnings and add Visual Studio Code files to gitignore

### PR DESCRIPTION
**Pull request by zebp**
I went around fixing some compiler warnings to keep things nice and tidy. There we a lot of raw types, so I gave them the most appropriate type I could.

### CODE REVIEW

**Review by shedaniel — APPROVED** [review]
Nice pr, has this been checked by checkstyle?

**Comment by shedaniel** [comment]
Just built in jitpack, seems like there are a few checkstyle violations: <https://jitpack.io/com/github/vlakreeh/patchwork-patcher/9b08b93644/build.log>

**Comment by zebp** [comment]
Woops, it totally slipped my mind! I'll get on that now.

**Comment by zebp** [comment]
Fixed the style inconsistencies.

**Review by shedaniel — APPROVED** [review]

**Review by ramidzkh — APPROVED** [review]

File: src/main/java/com/patchworkmc/PatchworkUI.java [hunk 1]
```diff
@@ -659,13 +662,14 @@ public static InputStream loadOrDownloadIntermediary(String version, File parent
 		return new FileInputStream(file);
 	}
 
+	// This class has a few unused fields at the moment, so I commented them out.
 	private static class YarnBuild {
 		String gameVersion;
-		String separator;
-		int build;
+		// String separator;
+		// int build;
```
Comment (natanfudge) [inline src/main/java/com/patchworkmc/PatchworkUI.java]: Have you tested this doesn’t break gson?

**Review by coderbot16 — CHANGES_REQUESTED** [review]
Just want the comment from Fudge to be addressed and then I'll merge this. Thanks!

File: src/main/java/com/patchworkmc/PatchworkUI.java (hunk 1 above)
Comment (shedaniel) [reply src/main/java/com/patchworkmc/PatchworkUI.java]: gson isn't strict like jotlin serialisation, even tho I don't really want these to be commented out.

Comment (zebp) [reply src/main/java/com/patchworkmc/PatchworkUI.java]: I didn't catch that when reading for through the source, I just addressed it and added the fields back.

**Review by coderbot16 — APPROVED** [review]
Thanks for this!

**Pull request merged**